cmd/abapEnvironmentCreateTag.go

```diff
@@ -0,0 +1,246 @@
+	if errorCookieJar != nil {
+		return errors.Wrap(errorCookieJar, "Could not create a Cookie Jar")
+	}
+
+	client.SetOptions(piperhttp.ClientOptions{
+		MaxRequestDuration: 180 * time.Second,
+		CookieJar:          cookieJar,
+		Username:           connectionDetails.User,
+		Password:           connectionDetails.Password,
+	})
+
+	backlog, errorPrepare := prepareBacklog(config)
+	if errorPrepare != nil {
+		return fmt.Errorf("Something failed during the tag creation: %w", errorPrepare)
+	}
+
+	err := createTags(backlog, telemetryData, connectionDetails, client, com)
+
+	return err
+}
+
+func createTags(backlog []CreateTagBacklog, telemetryData *telemetry.CustomData, con abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, com abaputils.Communication) (err error) {
+
+	connection := con
+	connection.XCsrfToken = "fetch"
+	connection.URL = con.URL + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Tags"
+	resp, err := abaputils.GetHTTPResponse("HEAD", connection, nil, client)
+	if err != nil {
+		err = abaputils.HandleHTTPError(resp, err, "Authentication on the ABAP system failed", con)
+		return err
+	}
+	defer resp.Body.Close()
+
+	log.Entry().WithField("StatusCode", resp.Status).WithField("ABAP Endpoint", connection.URL).Debug("Authentication on the ABAP system successful")
+	connection.XCsrfToken = resp.Header.Get("X-Csrf-Token")
+
+	errorOccurred := false
+	for _, item := range backlog {
+		err = createTagsForSingleItem(item, telemetryData, connection, client, com)
+		if err != nil {
+			errorOccurred = true
+		}
+	}
+
+	if errorOccurred {
+		message := "At least one tag has not been created"
+		log.Entry().Errorf(message)
+		return errors.New(message)
+	}
+	return nil
+
+}
+
+func createTagsForSingleItem(item CreateTagBacklog, telemetryData *telemetry.CustomData, con abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, com abaputils.Communication) (err error) {
+
+	for index := range item.tags {
+		err = createSingleTag(item, index, telemetryData, con, client, com)
+	}
+	return err
+}
+
+func createSingleTag(item CreateTagBacklog, index int, telemetryData *telemetry.CustomData, con abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, com abaputils.Communication) (err error) {
+
+	requestBodyStruct := CreateTagBody{RepositoryName: item.repositoryName, CommitID: item.commitID, Tag: item.tags[index].tagName, Description: item.tags[index].tagDescription}
+	errorMessage := "Could not create tag " + requestBodyStruct.Tag + " for repository " + requestBodyStruct.RepositoryName + " with commitID " + requestBodyStruct.CommitID
+	requestBodyJson, err := json.Marshal(&requestBodyStruct)
+	if err != nil {
+		return err
+	}
+
+	log.Entry().Debugf("Request body: %s", requestBodyJson)
+	resp, err := abaputils.GetHTTPResponse("POST", con, requestBodyJson, client)
+	if err != nil {
+		err = abaputils.HandleHTTPError(resp, err, errorMessage, con)
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Parse response
+	var body CreateTagResponse
```

Comment:
this isn' the body, it's the createTagResponse, or?

Reply:
the response body of the create tag response